apisix/plugins/batch-requests.lua

```diff
@@ -0,0 +1,239 @@
+                                "OPTIONS", "CONNECT", "TRACE"},
+                        default = "GET"
+                    },
+                    path = {
+                        type = "string",
+                        minLength = 1,
+                    },
+                    query = {
+                        description = "request header",
+                        type = "object",
+                    },
+                    headers = {
+                        description = "request query string",
+                        type = "object",
+                    },
+                    ssl_verify = {
+                        type = "boolean",
+                        default = false
+                    },
+                }
+            }
+        }
+    },
+    anyOf = {
+        {required = {"pipeline"}},
+    },
+}
+
+local _M = {
+    version = 0.1,
+    priority = 4010,
+    name = plugin_name,
+    schema = schema
+}
+
+
+function _M.check_schema(conf)
+    local ok, err = core.schema.check(schema, conf)
+    if not ok then
+        return false, err
+    end
+    return true
+end
+
+
+local function check_input(data)
+    local ok, err = core.schema.check(req_schema, data)
+    if not ok then
+        return 400, {error_msg = "bad request body: " .. err}
+    end
+end
+
+
+local function set_common_header(data)
+    if not data.headers then
+        return
+    end
+
+    for i,req in ipairs(data.pipeline) do
+        if not req.headers then
+            req.headers = data.headers
+        else
+            for k, v in pairs(data.headers) do
+                if not req.headers[k] then
+                    req.headers[k] = v
+                end
+            end
+        end
+    end
+end
+
+
+local function set_common_query(data)
+    if not data.query then
+        return
+    end
+
+    for i,req in ipairs(data.pipeline) do
+        if not req.query then
+            req.query = data.query
+        else
+            for k, v in pairs(data.query) do
+                if not req.query[k] then
+                    req.query[k] = v
+                end
+            end
+        end
+    end
+end
+
+
+local function get_file(file_name)
+    local f = io_open(file_name, 'r')
+    if f then
+        local req_body = f:read("*all")
+        f:close()
+        return req_body
+    end
+
+    return
+end
+
+
+local function batch_requests()
+    ngx.req.read_body()
+    local req_body = ngx.req.get_body_data()
+    if not req_body then
+        local file_name = ngx.req.get_body_file()
+        if file_name then
+            req_body = get_file(file_name)
+        end
+
+        if not req_body then
+            core.response.exit(400, {
+                message = "no request body, you should give at least one pipeline setting"
+            })
+        end
+    end
+
+    local data, err = core.json.decode(req_body)
+    if not data then
+        core.response.exit(400, {message = "invalid request body", req_body = req_body, err = err})
```

Comment:
we should use only one style, eg: `{error_msg = "**** " .. err}`, it is easier for the client shows the error message.

here is the example: https://github.com/apache/incubator-apisix/pull/1388/files#diff-7a39e1834cde089a2da404821d733a92R111

please fix the other similar points.